### PR TITLE
Ethan se/5

### PR DESCRIPTION
**Pull request by ethanSE**
implement #5 

Adds state, logic, UI for selecting a piece to play from a player's hand.

Can only place a selected piece.

Can only select an available piece.

The currently selected piece is highlighted in hand. 

Previously played pieces are grey and not interactable.

Can only select one piece at a time.

Can only select piece during your turn.

The currently selected piece is cleared on turn pass.

Selected piece is cleared and marked as unavailable when played.

### CODE REVIEW

File: src/components/ActivePlayer.tsx [hunk 1]
```diff
@@ -3,20 +3,21 @@ import { useContext } from 'react';
 import GameStateContext from '../context/GameStateContext';
```
Comment (ethanSE) [inline src/components/ActivePlayer.tsx:3]: every change here is a refactor to use css modules

File: src/spec/PlayerHand.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    const selected = hand.setSelectedPiece('piece1');
+    expect(selected.getSelectedPiece()).toStrictEqual(O.some({ name: 'piece1', status: 'selected' }))
+
+    const selected2 = selected.setSelectedPiece('piece4');
+    expect(selected2.getSelectedPiece()).toStrictEqual(O.some({ name: 'piece4', status: 'selected' }))
+
+    const cleared = selected2.clearSelectedPiece();
+    expect(cleared.getSelectedPiece()).toBe(O.none)
+});
```
Comment (ethanSE) [inline src/spec/PlayerHand.test.ts:16]: I could have added some more tests.
I'm expecting these classes to change and haven't yet decided on interfaces that these should implement.
These tests are a bit brittle and the classes currently are a bit of a leaky abstraction.
Once I get into the logic and representation for the pieces, board, and game rules I should finalize some interfaces for these and write better tests to the implementation of the interfaces as opposed to the internal representation

**Review by leblancPDX — APPROVED** [review]
LGTM

I did not have as much time to review this as I might have preferred, but decided it was best to approve and let you move on.

**Comment by ethanSE** [comment]
> LGTM
> 
> I did not have as much time to review this as I might have preferred, but decided it was best to approve and let you move on.

Thanks for taking a look. It was a lot of changes

**Pull request merged**